contrib: Install modprobe to cilium-runtime image

brb opened the pull request:
This PR changes the way we make `modprobe` accessible to `cilium-agent` running in a container. Previously, we have bind-mounted `/sbin/modprobe` to the container. This approach had the following drawbacks:

* `modprobe` can be installed on a host in a different location than `/sbin`.
* `modprobe` can be dynamically-linked which might fail when running from a container. E.g.: `modprobe[10018]: segfault at 1a010 ip 00007f31ba4598b8 sp 00007fff6848d1d8 error 4 in libc-2.27.so[7f31ba2ef000+1e7000]` (observed on GKE).

Instead, distribute `modprobe` via the `kmod` package in the `cilium-runtime` container image.

Fixes: e45c4b2 ("datapath/iptables: Check iptables kernel modules")
Fixes: 464978c0 ("k8s: Add SYS_MODULE capability, module paths")

**OLD PR DESCRIPTION**: ~Currently, the cilium-agent relies on `modprobe` to install missing kernel modules. The `kmod` package for Ubuntu installs the `modprobe` executable and we currently do not ship it with the Cilium container image which can lead to the following error (reported by a user):~

```
IPv6 is enabled and ip6tables modules could not be initialized" error="could not load module ip6_tables: exec: \"modprobe\": executable file not found in $PATH" subsys=iptables
```

---

~Once this merged, we need to trigger a new build for `cilium-runtime` and then start using it in the main Dockerfile.~ DONE.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/7820)
<!-- Reviewable:end -->

brb committed:
contrib: Install modprobe to cilium-runtime image

Currently, the cilium-agent relies on `modprobe` to install missing
kernel modules. The `kmod` package for Ubuntu installs the `modprobe`
executable and we currently do not ship it with the Cilium container
image which can lead to the following error (reported by a user):

```
IPv6 is enabled and ip6tables modules could not be initialized" error="could not load module ip6_tables: exec: \"modprobe\": executable file not found in $PATH" subsys=iptables
```

Fixes: e45c4b2 ("datapath/iptables: Check iptables kernel modules")
Signed-off-by: Martynas Pumputis <m@lambda.lt>

brb committed:
Update cilium-runtime image

The updated image includes `modprobe`.

Signed-off-by: Martynas Pumputis <m@lambda.lt>

vadorovsky reviewed (approved):
Thanks for the fix and sorry!

brb committed:
examples: Do not bind mount /sbin/modprobe

The host's modprobe might be dynamically linked which running
from a container might not work as expected. E.g.:

[ 1112.449703] modprobe[10018]: segfault at 1a010 ip 00007f31ba4598b8 sp
00007fff6848d1d8 error 4 in libc-2.27.so[7f31ba2ef000+1e7000]

(Experienced on GKE)

Fixes: 464978c0 ("k8s: Add SYS_MODULE capability, module paths")
Signed-off-by: Martynas Pumputis <m@lambda.lt>
